### DISCUSSION TITLE
Jamulus 3.11.0beta1 release

### DISCUSSION THREAD

**Discussion by pljones** [jamulussoftware/jamulus#3342] [Announcements]
I am very happy to announce the first beta for Jamulus 3.11.0, [Jamulus 3.11.0beta1](https://github.com/jamulussoftware/jamulus/releases/tag/r3_11_0beta1) (version reported is just "beta", not "beta1"...).  As of now, we have stabilised what will change in the final release and will only be no further changes - *unless bugs are found*, of course.

And that's where everyone here can help!

Please download the installer for your platform from the [release note](https://github.com/jamulussoftware/jamulus/releases/tag/r3_11_0beta1) and start using it for your normal usage of Jamulus - **but keep a back-up of your 3.10.0 install** for now.

**AS OF NOW, THE MAC OS RELEASE REMAINS UNSIGNED**. This beta will remain unsigned. We plan to get a signed version ready in a later beta or release candidate.

If you find any problems, please raise a bug report making it clear that you're using the **3.11.0beta1** release.

For people who have to build from source, please note the dependencies have changed, the one major change being the minimum version of Qt is now 5.12.2.  (Builds have been with Qt up to 6.7.2.)

The updated website will be release formally once all the translations are complete and the final release is announced.  For now, there's [a snapshot of the website hosted here](http://jamuluswebsite.drealm.info).

**Comment by rdica** [comment]
Testing client (Ubuntu Studio 22.04) and server (Ubuntu Server 24.04).
Adding and removing servers with button in Connect dialog has no issues.
Adding and removing Directory servers with button in Settings dialog has no issues.
Stopped and started client after each add, then again for each removal and looked at ini file to confirm adds/removals. Looks good.
Server up and running on Any Genre2, no issues so far.

**Comment by rdica** [comment]
While not an error or warning, its behaviour I haven't seen previously when building client or server:
```
qmake "CONFIG+=noupcasename headless serveronly" Jamulus.pro
Info: creating stash file /home/jammer/src/jamulus-r3_11_0beta1-clean/jamulus-r3_11_0beta1/.qmake.stash
Project MESSAGE: The target name is jamulus instead of Jamulus.
Project MESSAGE: building version "3.11.0beta" (release)
Project MESSAGE: Headless mode activated.
Project MESSAGE: Restricting build to server-only due to CONFIG+=serveronly.
Project MESSAGE: server only, skipping client rpc
Project MESSAGE: The target name is jamulus instead of Jamulus.
Project MESSAGE: building version "3.11.0beta" (release)
Project MESSAGE: Headless mode activated.
Project MESSAGE: Restricting build to server-only due to CONFIG+=serveronly.
Project MESSAGE: server only, skipping client rpc
Project MESSAGE: The target name is jamulus instead of Jamulus.
Project MESSAGE: building version "3.11.0beta" (release)
Project MESSAGE: Headless mode activated.
Project MESSAGE: Restricting build to server-only due to CONFIG+=serveronly.
Project MESSAGE: server only, skipping client rpc
```
Looks like the project message is output three times.
Double checked a clean untar of `jamulus-r3_10_0.tar.gz` and get:
```
qmake "CONFIG+=noupcasename headless serveronly" Jamulus.pro
Info: creating stash file /home/jammer/src/jamulus-r3_10_0-clean/jamulus-r3_10_0/.qmake.stash
Project MESSAGE: The target name is jamulus instead of Jamulus.
Project MESSAGE: building version "3.10.0" (release)
Project MESSAGE: Headless mode activated.
Project MESSAGE: Restricting build to server-only due to CONFIG+=serveronly.
Project MESSAGE: server only, skipping client rpc
```
Client build:
```
qmake "CONFIG+=noupcasename" Jamulus.pro
Info: creating stash file /home/jammer/src/jamulus-r3_11_0beta1-clean/jamulus-r3_11_0beta1/.qmake.stash
Project MESSAGE: The target name is jamulus instead of Jamulus.
Project MESSAGE: building version "3.11.0beta" (release)
Project MESSAGE: JACK Audio Interface Enabled.
Project MESSAGE: The target name is jamulus instead of Jamulus.
Project MESSAGE: building version "3.11.0beta" (release)
Project MESSAGE: JACK Audio Interface Enabled.
Project MESSAGE: The target name is jamulus instead of Jamulus.
Project MESSAGE: building version "3.11.0beta" (release)
Project MESSAGE: JACK Audio Interface Enabled.
```
Build OSes: Ubuntu Server 24.04, Ubunto Studio 22.04.

**Reply by pljones** [reply]
I'd guess it's this change in `Jamulus.pro`:
```
@@ -22,7 +28,9 @@ contains(VERSION, .*dev.*) {

 CONFIG += qt \
     thread \
-    lrelease
+    lrelease \
+    embed_translations \
+    debug_and_release

 QT += network \
     xml \
@@ -42,9 +50,8 @@ contains(CONFIG, "headless") {
     QT += multimedia
 }
```
@softins can you confirm it's expected behaviour?

**Reply by softins** [reply]
Yes it is expected behaviour. Prior to the above change, `qmake` was just generating `Makefile`. Now instead it generates `Makefile`, `Makefile.Release` and `Makefile.Debug`. One or other of the latter two is called from `Makefile` depending on whether the user does `make`, `make release` (the default, the same as `make`) or `make debug`.

**Comment by chrisrimple** [comment]
![image](https://github.com/user-attachments/assets/d9e7fa98-6f2e-4250-ac5d-6d23e412f874)

Not sure if this is a "bug", but the default window size for the Mac installer (DMG) FOR 3.10 is too small and requires vertical scrolling.  I haven't tested on 3.11 but suspect the problem exists.

**Reply by ann0see** [reply]
I think I saw that before if tabs (?) are enabled in finder?

**Reply by pljones** [reply]
Can we get this as a bug report if it's a regression since 3.10.0?

(Could it be related to any tooling changes with the Mac builds?)

**Reply by ann0see** [reply]
I didn't see it on my build. It's an UI issue which probably shows up due to scaling.

**Comment by dcorson-ticino-com** [comment]
Have been using the client under Win10 for the last 2 weeks and see no problems.

**Comment by pljones** [comment]
Beta 2 is now out, closing.
